photon-lib/src/main/scala/com/linkedin/photon/ml/evaluation/EvaluationSuite.scala

```diff
@@ -68,10 +70,14 @@ class EvaluationSuite(
     val evaluations = evaluators
       .map {
         case evaluator: SingleEvaluator =>
-          (evaluator.evaluatorType, evaluator.evaluate(scoresAndLabelsAndWeights.values))
+          (evaluator.evaluatorType, (evaluator.evaluate(scoresAndLabelsAndWeights.values), None))
 
         case multiEvaluator: MultiEvaluator =>
-          (multiEvaluator.evaluatorType, multiEvaluator.evaluate(scoresAndLabelsAndWeights))
+          if (savePerGroupEvaluation) {
+            (multiEvaluator.evaluatorType, multiEvaluator.evaluatePerGroup(scoresAndLabelsAndWeights))
```

Comment:
What if we need to train multiple random effects? For example we want to first train per-member and then train per-job models based on the residuals from per-member models. Previously we get metrics from per-member models only as well as metrics from per-member + per-job models. If we want to know member-level metrics but per-member models train first, we need to get the final score and choose `member` instead of `job` as the group. Is it implemented in this way?

Reply:
This is a good question. I think no matter of the order of random effect training, the final per-member level metric is computed by grouping on the member level, same for per-job metric. That's the same as computing the average of per-member metric previously.